### PR TITLE
Setup mkdocs

### LINKED ISSUE #2117: Set up documentation
In the `[Mathesar Core] New documentation` email thread, we discussed a setup for hosting documentation.

This task is about implementing a documentation setup based on that. IIRC, we've not speced it out fully, so the email thread, as it is now, is a starting point for the spec.

### PR DESCRIPTION
**Pull request by dmos62**
Fixes #2117

Sets up `mkdocs`.

Adds two things:
- `docs` directory that holds a fresh `mkdocs` project;
- `deploy-docs` github workflow that:
  - when `master` branch is pushed to, builds the docs inside `docs`, and puts them inside `gh-pages` branch.

We'll want to add a custom domain and maybe only publish documentation for releases, but that's outside the scope of this PR.

I left (practically) everything on default. The only liberties I took are:
- using `mkdocs-material`, instead of plain `mkdocs`, since it's a bit nicer (it's a popular `mkdocs` theme/distribution with batteries included);
- deploying using `mkdocs gh-deploy` instead of `mkdocs gh-deploy --force`: somehow didn't feel right using `--force` by default: here's [the tutorial I followed](https://squidfunk.github.io/mkdocs-material/publishing-your-site/).

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by kgodey** [comment]
Once this is merged, please create an issue for me to:
- Set up the `docs.mathesar.org` subdomain
- Set up GitHub Pages deployment in this repo

**Comment by kgodey** [comment]
I see you're already on it, thanks!

**Review by seancolsen — COMMENTED** [review]
A few thoughts:

- I've never used `mkdocs gh-deploy` before, but one thing I know about deploying static sites is this: it's best to err on the side of aborting deployment if anything goes wrong during build (rather than risk deploying a partially-built site). I've run into this specifically with mkdocs in the past and (although my memory is hazy) I believe I fixed it by passing the `--strict` flag to `mkdocs build`. I see that the `gh-deploy` command has a `--strict` flag too, so I think it would be wise to enable it.

- It would be nice to add some meta-docs to help devs write documentation as they're coding:

    - A new `README.md` within the `docs` directory, explaining how to install mkdocs and launch it locally. I do think it makes sense to recommend that people run mkdocs on their host machine (instead of through Docker), though we could debate that.
    - A very small section within the root-level repo `README.md`, directing readers to the README within the `docs` dir.

- I made a number of [suggestions](https://groups.google.com/a/mathesar.org/g/core-team/c/yMOA2b7IRJs/m/Cpxx7o0_AwAJ) via email. I see we've adopted some of them here, like using mkdocs-material. I do think it would be good to enable my recommended markdown extensions in this PR too, that way we can begin using that special markdown right away.

**Comment by dmos62** [comment]
@seancolsen thanks, I'll implement your suggestions.

**Comment by dmos62** [comment]
@seancolsen can you re-review?

- I added and linked to a `docs/README.md` with links to various references and CiviCRM's style guide;
- added the mentioned markdown extensions;
- added the redirection plugin (and linked to its homepage in the README);
- added the `--strict` flag to `mkdocs gh-deploy` in the github workflow action.

**Comment by dmos62** [comment]
Removed @silentninja as reviewer (one reviewer is enough) and re-requested review from @seancolsen.

File: docs/README.md
Comment (seancolsen) [inline docs/README.md:9]: I predict that we may need to flesh these instructions out at some point, providing some clearer steps, akin to these [steps that I wrote for CiviCRM](https://docs.civicrm.org/dev/en/latest/documentation/#mkdocs). But we can see if anyone gets confused and add more clarity later if needed.

**Review by seancolsen — APPROVED** [review]
This is looking good @dmos62. I'll leave it to you to merge, since I'm not totally clear on the scope of the ticket. Obviously we also need to write some docs _content_, but that can come later.

**Pull request merged**
